core/src/main/java/com/themecleanflex/models/ReferenceModel.java

```diff
@@ -609,14 +606,13 @@ private String generateReferenceJson() {
     private Map findNode(Map map, String name, String value) {
       for (Object key : map.keySet()) {
         Object val = map.get(key);
-        if(key.equals(name)) {
-          if(value.equals(val)) {
+        if(equals(key, name)) {
+          if(equals(value, val)) {
             return map;
           }
         }
-        if(key.equals("children") && val instanceof ArrayList) {
+        if("children".equals(key) && val instanceof ArrayList) {
```

Comment:
is there a reason why you use `equals(obj, other)` above but `String.equals(otherString)` here?